tests/daemon/test-aggregate-tally.c

```diff
@@ -0,0 +1,194 @@
+#define TEST_DIRECTORY "/tmp/aggregate-tally"
+
+// Generated with uuidgen
+
+static const char *uuids[] = {
+  "41d45e08-5e72-4c43-8cbf-ef37bb4411a4",
+  "03177773-7513-4866-ae97-bb935f2c5384",
+  "a692ce9c-8684-4d6b-97d5-07f39e0a8560",
+};
+
+struct AggregateEvent
+{
+  guint32 unix_user_id;
+  GVariant *event_id;
+  GVariant *aggregate_key;
+  GVariant *payload;
+};
+
+struct Fixture
+{
+  EmerAggregateTally *tally;
+  GFile *tally_folder;
+};
+
+static GVariant *
+event_id_to_variant (const char *event_id)
+{
+  GVariantBuilder event_id_builder;
+  uuid_t parsed_event_id;
+
+  if (uuid_parse (event_id, parsed_event_id) != 0)
+    return NULL;
+
+  g_variant_builder_init (&event_id_builder, G_VARIANT_TYPE ("ay"));
+  for (size_t i = 0; i < UUID_LENGTH; ++i)
+    g_variant_builder_add (&event_id_builder, "y", parsed_event_id[i]);
+
+  return g_variant_new ("(aymv)", &event_id_builder, NULL);
+}
+
+static struct AggregateEvent *
+create_aggregate_event (guint32     unix_user_id,
+                        const char *event_id,
+                        GVariant   *aggregate_key,
+                        GVariant   *payload)
+{
+  struct AggregateEvent *event;
+
+  event = g_new0 (struct AggregateEvent, 1);
+  event->unix_user_id = unix_user_id;
+  event->event_id = g_variant_ref_sink (event_id_to_variant (event_id));
+  event->aggregate_key = g_variant_ref_sink (aggregate_key);
+  event->payload = g_variant_ref_sink (payload);
+
+  return event;
+}
+/*
+static gboolean
+check_file_exists (const char *date,
+                   const char *filename)
+{
+  g_autofree char *path = NULL;
+
+  path = g_build_filename (TEST_DIRECTORY,
+                           date,
+                           filename,
+                           NULL);
+
+  return g_file_test (path, G_FILE_TEST_EXISTS);
+}
+*/
+static void
+teardown (struct Fixture *fixture,
+          gconstpointer   dontuseme)
+{
+  g_clear_object (&fixture->tally);
+
+  if (fixture->tally_folder)
+    {
+      delete_folder (fixture->tally_folder, NULL, NULL);
+      g_clear_object (&fixture->tally_folder);
+    }
+}
+
+static void
+setup (struct Fixture *fixture,
+       gconstpointer   dontuseme)
+{
+  teardown (fixture, dontuseme);
+
+  g_mkdir (TEST_DIRECTORY, 0777);
+  fixture->tally_folder = g_file_new_for_path (TEST_DIRECTORY);
+  fixture->tally = emer_aggregate_tally_new (TEST_DIRECTORY);
+}
+
+// Testing Cases
+
+static void
+test_aggregate_tally_new_succeeds (struct Fixture *fixture,
+                                   gconstpointer   dontuseme)
+{
+  g_assert_nonnull (fixture->tally);
+}
+
+static void
+test_aggregate_tally_store_single_event (struct Fixture *fixture,
+                                         gconstpointer   dontuseme)
+{
+  struct AggregateEvent *event;
+
+  event = create_aggregate_event (0,
+                                  uuids[0],
+                                  g_variant_new ("v", g_variant_new_string ("")),
```

Comment:
I don't think these need to be GVariants containing a single variant, they can be:

```suggestion
                                  g_variant_new_string ("")
```